perses/annihilation/relative.py

```diff
@@ -2629,3 +2629,2206 @@ def _handle_exceptions(self):
+        "old_charge_scalar1 = lambda_electrostatics_old * charge_old1;",
+        "new_charge_scalar1 = lambda_electrostatics_new * charge_new1;",
+
+        "charge2 = (is_unique_old2 * old_charge_scalar2) + (is_unique_new2 * new_charge_scalar2) + is_core2 * (old_charge_scalar2 + new_charge_scalar2) + is_environment2 * (old_charge_scalar2 + new_charge_scalar2);",
+        "old_charge_scalar2 = lambda_electrostatics_old * charge_old2;",
+        "new_charge_scalar2 = lambda_electrostatics_new * charge_new2;",
+
+        # Define sigma
+        "sigma = (sigma1 + sigma2) / 2;",
+
+        # Define sigma1 and sigma2 (with alchemical scaling)
+        "sigma1 = (is_unique_old1 * old_sigma_scalar1) + (is_unique_new1 * new_sigma_scalar1) + is_core1 * (old_sigma_scalar1 + new_sigma_scalar1) + is_environment1 * (old_sigma_scalar1 + new_sigma_scalar1);",
+        "old_sigma_scalar1 = lambda_sterics_old * sigma_old1;",
+        "new_sigma_scalar1 = lambda_sterics_new * sigma_new1;",
+
+        "sigma2 = (is_unique_old2 * old_sigma_scalar2) + (is_unique_new2 * new_sigma_scalar2) + is_core2 * (old_sigma_scalar2 + new_sigma_scalar2) + is_environment2 * (old_sigma_scalar2 + new_sigma_scalar2);",
+        "old_sigma_scalar2 = lambda_sterics_old * sigma_old2;",
+        "new_sigma_scalar2 = lambda_sterics_new * sigma_new2;",
+
+        # Define epsilon (with rest scaling)
+        "epsilon = sqrt((epsilon1 * p1_rest_scale) * (epsilon2 * p2_rest_scale));",
+        # Q for DAR: Are we sure we want to apply the rest scaling before combining epsilons?
+
+        # Define epsilon1 and epsilon2 (with alchemical scaling)
+        "epsilon1 = (is_unique_old1 * old_epsilon_scalar1) + (is_unique_new1 * new_epsilon_scalar1) + is_core1 * (old_epsilon_scalar1 + new_epsilon_scalar1) + is_environment1 * (old_epsilon_scalar1 + new_epsilon_scalar1);",
+        "old_epsilon_scalar1 = lambda_sterics_old * epsilon_old1;",
+        "new_epsilon_scalar1 = lambda_sterics_new * epsilon_new1;",
+
+        "epsilon2 = (is_unique_old2 * old_epsilon_scalar2) + (is_unique_new2 * new_epsilon_scalar2) + is_core2 * (old_epsilon_scalar2 + new_epsilon_scalar2) + is_environment2 * (old_epsilon_scalar2 + new_epsilon_scalar2);",
+        "old_epsilon_scalar2 = lambda_sterics_old * epsilon_old2;",
+        "new_epsilon_scalar2 = lambda_seterics_new * epsilon_new2;",
+
+        # Define alpha
+        "alpha = sqrt(-log(2 * delta) / r_cutoff);",
+        "delta = {delta};",
+        "r_cutoff = {r_cutoff};",
+
+        # Define r_eff
+        "r_eff_electrostatics = sqrt(r^2 + w_electrostatics^2);",
+        "r_eff_sterics = sqrt(r^2 + w_sterics^2);",
+
+        # Define 4th dimension terms:
+        "w_electrostatics = is_unique_old * lambda_electrostatics_new * w_scale + is_unique_new * lambda_electrostatics_new * w_scale;", # Q for DAR: do we also want to multiply by r_cutoff (as we do in RxnHybridTopologyFactory)?
```

Comment:
i guess the w value should go from 0 to some >0 value, (or backward). but the size of the scaling factor is an open parameter. i guess we can multiply it by `w_scale` * `r_cutoff` where `w_scale` should be a scalar between 0 and 1?

Reply:
default w_scale to 0.1

Reply:
fixed